lib/ansible/module_utils/network/iosxr/utils/utils.py

```diff
@@ -61,3 +62,188 @@ def dict_delete(base, comparable):
 
 def pad_commands(commands, interface):
     commands.insert(0, 'interface {0}'.format(interface))
+
+
+def remove_command_from_config_list(interface, cmd, commands):
+    # To delete the passed config
+    if interface not in commands:
+        commands.insert(0, interface)
+    commands.append('no %s' % cmd)
+    return commands
+
+
+def add_command_to_config_list(interface, cmd, commands):
+    # To set the passed config
+    if interface not in commands:
+        commands.insert(0, interface)
+    commands.append(cmd)
+
+
+def dict_diff(sample_dict):
+    # Generate a set with passed dictionary for comparison
+    test_dict = {}
+    for k, v in iteritems(sample_dict):
+        if v is not None:
+            if isinstance(v, list):
+                if isinstance(v[0], dict):
+                    li = []
+                    for each in v:
+                        for key, value in iteritems(each):
+                            if isinstance(value, list):
+                                each[key] = tuple(value)
+                        li.append(tuple(each.items()))
+                    v = tuple(li)
+                else:
+                    v = tuple(v)
+            elif isinstance(v, dict):
+                li = []
+                for key, value in iteritems(v):
+                    if isinstance(value, list):
+                        v[key] = tuple(value)
+                li.extend(tuple(v.items()))
+                v = tuple(li)
+            test_dict.update({k: v})
+    return_set = set(tuple(test_dict.items()))
+    return return_set
+
+
+def filter_dict_having_none_value(want, have):
```

Comment:
Can you have a look at the `dict_delete` method which I already added in utils? https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/iosxr/utils/utils.py#L35

Reply:
It's not exactly the same, coz it checks for want none value which is present and configured in have, and this checks for all the input type be it the element of list/dict.